hmpy/views/switch.py

```
@@ -0,0 +1,147 @@
+            paint.drawRect(-5 * scale_x, 10 * scale_y,
+                           55 * scale_x, 70 * scale_y)
+
+            if self.isChecked():
+                paint.setBrush(Qt.green)
+                paint.drawRect(-5 * scale_x, 10 * scale_y,
+                               55 * scale_x, 35 * scale_y)
+                paint.setBrush(Qt.darkGreen)
+                paint.drawRect(-5 * scale_x, 8 * scale_y,
+                               55 * scale_x, 10 * scale_y)
+            else:
+                paint.setBrush(Qt.red)
+                paint.drawRect(-5 * scale_x, 45 * scale_y,
+                               55 * scale_x, 35 * scale_y)
+
+                paint.setBrush(Qt.darkRed)
+                paint.drawRect(-5 * scale_x, 70 * scale_y,
+                               55 * scale_x, 10 * scale_y)
+
+            paint.save()
+
+            paint.translate(self.width() / 2, 35 * scale_y)
+
+            paint.setBrush(Qt.darkGray)
+            paint.drawRect(-70 * scale_x, 50 * scale_y,
+                           85 * scale_x, 12 * scale_y)
+
+            paint.setPen(Qt.white)
+
+            paint.drawText(-55 * scale_x, 58 * scale_y, self._text)
+
+            paint.restore()
+            paint.end()
+
+            # Convert to pixmap and return
+            return QPixmap.fromImage(image)
+
+    def __init__(self, text, on=False):
+        """Initialize the switch view.
+
+        :param on: the state of the button"""
+        super().__init__()
+        self._button = self._SwitchButton(text, self)
+        self._button.setChecked(on)
+
+    @property
+    def on(self):
+        return self._button.isChecked()
+
+    @on.setter
+    def on(self, state):
+        """
+        Set the button state
+
+        :param state:
+        :return:
+        """
+        if state != self._button.isChecked():
+            self._button.toggle()
+
+    def toggle(self):
+        self._button.toggle()
+
+    def on_toggle(self, callback):
```

Should probably throw an exception otherwise